File: android/app/src/main/java/com/laneshadow/ui/components/molecules/ConnectionBanner.kt
package com.laneshadow.ui.components.molecules

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.laneshadow.theme.LocalLaneShadowTheme

/**
 * ConnectionBanner molecule component
 *
 * Network status banner displayed when connection is required or limited.
 * Following React Native wrapper patterns from react-native/components/ui/connection-banner.tsx
 *
 * @param message Banner message text (default: "Connection Required - Some features may be limited")
 * @param isVisible Whether to show the banner (default: true)
 * @param modifier Modifier for the banner container
 */
@Composable
fun ConnectionBanner(
    message: String = "Connection Required - Some features may be limited",
    isVisible: Boolean = true,
    modifier: Modifier = Modifier,
) {
    if (!isVisible) {
        return
    }

    val theme = LocalLaneShadowTheme.current

    // Background color from theme warning token
    val backgroundColor = theme.colors.warning.default

    // Text color from theme onPrimary token with fallback to white
    val textColor = theme.colors.onPrimary.default

    // Build semantics with accessibility description
    val bannerModifier = modifier.semantics {
        this.contentDescription = "Network status warning"
    }

    Surface(
        modifier = bannerModifier,
        color = backgroundColor,
    ) {
        Row(
            modifier = Modifier
                .padding(all = theme.space.md),
            horizontalArrangement = Arrangement.Center,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            // Warning icon
            Icon(
                imageVector = Icons.Default.Warning,
                contentDescription = null, // Decorative icon, description is on parent
                tint = textColor,
                modifier = Modifier.size(18.dp),
            )

            // Message text
            Text(
                text = message,
                style = androidx.compose.ui.text.TextStyle(
                    fontSize = 14.sp,
                    fontWeight = androidx.compose.ui.text.font.FontWeight.Medium,
                ),
                color = textColor,
                modifier = Modifier.padding(start = 8.dp),
            )
        }
    }
}

File: ios/LaneShadow/Views/Molecules/ConnectionBanner.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Connection Banner Component

/**
 * Connection banner molecule component
 *
 * Displays a full-width warning banner for network connection status.
 * Following React Native component from react-native/components/ui/connection-banner.tsx
 *
 * ## Design Tokens Used
 * - Colors:
 *   - Background: `theme.colors.warning.default`
 *   - Text: `Color.white` (onPrimary equivalent)
 * - Layout:
 *   - Width: `.infinity` (full-width)
 *   - Padding: 12pt all sides (theme.space.md equivalent)
 * - Typography:
 *   - System font, 14pt (bodySmall equivalent), medium weight
 * - Icon:
 *   - System wifi icon (20pt)
 *
 * ## Parameters
 * - message: Banner message text (default: "Connection Required - Some features may be limited")
 * - isVisible: Show/hide the banner (default: true)
 *
 * ## Usage
 * ```swift
 * LSConnectionBanner()
 * LSConnectionBanner(message: "Offline mode - Changes will sync when connected")
 * LSConnectionBanner(isVisible: hasNetworkIssue)
 * ```
 */
public struct LSConnectionBanner: View {
    @Environment(\.theme) private var theme

    private let message: String
    private let isVisible: Bool

    public init(
        message: String = "Connection Required - Some features may be limited",
        isVisible: Bool = true
    ) {
        self.message = message
        self.isVisible = isVisible
    }

    // MARK: - Body

    public var body some View {
        if isVisible {
            HStack(spacing: 8) {
                // Wifi/exclamation icon on the left
                Image(systemName: "wifi.exclamationmark")
                    .font(.system(size: 20, weight: .medium))
                    .foregroundStyle(Color.white)

                // Message text
                Text(message)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundStyle(Color.white)
                    .frame(maxWidth: .infinity, alignment: .center)
            }
            .frame(maxWidth: .infinity)
            .padding(12)
            .background(theme.colors.warning.default)
            .accessibilityElement(children: .combine)
            .accessibilityLabel("Connection status: \(message)")
        }
    }
}

// MARK: - Preview

#Preview("ConnectionBanner - Default") {
    VStack(spacing: 0) {
        LSConnectionBanner()

        Spacer()

        Text("App content goes here")
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(Color.gray.opacity(0.1))
    }
    .laneShadowTheme()
}

#Preview("ConnectionBanner - Custom Message") {
    VStack(spacing: 0) {
        LSConnectionBanner(message: "Offline mode - Changes will sync when connected")

        Spacer()

        Text("App content goes here")
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(Color.gray.opacity(0.1))
    }
    .laneShadowTheme()
}

#Preview("ConnectionBanner - Hidden") {
    VStack(spacing: 0) {
        LSConnectionBanner(isVisible: false)

        Spacer()

        Text("Banner is hidden - app content shows normally")
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(Color.gray.opacity(0.1))
    }
    .laneShadowTheme()
}

#Preview("ConnectionBanner - Dark Mode") {
    VStack(spacing: 0) {
        LSConnectionBanner()

        Spacer()

        Text("Dark mode content")
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(Color.black)
    }
    .laneShadowTheme()
    .preferredColorScheme(.dark)
}
